h2integrate/storage/generic_storage_cost.py

```diff
@@ -0,0 +1,91 @@
+    capacity and charge rate for any commodity.
+
+    Attributes:
+        capacity_capex (float|int): storage energy capital cost in $/capacity_units
+        charge_capex (float|int): storage power capital cost in $/charge_units/h
+        opex_fraction (float): annual operating cost as a fraction of the total system cost.
+        cost_year (int): dollar year corresponding to input costs
+        max_capacity (float): Maximum storage capacity (in non-rate units,
+            e.g., "kW*h" if `commodity_units` is "kW").
+        max_charge_rate (float): Maximum rate at which storage can be charged (in units
+            per time step, e.g., "kW/time step").
+        commodity_units (str): Units of the storage resource used to define the charge rate.
+            max_capacity and max_charge_rate. Must have a base of Watts ('W') or grams ('g/h')
+            or heat ('MMBtu/h')
+    """
+
+    capacity_capex: float | int = field(validator=gte_zero)
+    charge_capex: float | int = field(validator=gte_zero)
+    opex_fraction: float = field(validator=range_val(0, 1))
+    max_capacity: float = field()
+    max_charge_rate: float = field()
+    commodity_units: str = field(
+        validator=contains(["W", "kW", "MW", "GW", "TW", "g/h", "kg/h", "t/h", "MMBtu/h"])
+    )
+
+
+class GenericStorageCostModel(CostModelBaseClass):
+    """Generic storage cost model for any commodity (electricity, hydrogen, etc.).
+
+    This model calculates costs based on storage capacity and charge/discharge rate.
+
+    Total_CapEx = capacity_capex * Storage_Hours + charge_capex
+
+    - Total_CapEx: Total System Cost (USD/charge_units)
+    - Storage_Hours: Storage Duration (hr)
+    - capacity_capex: Storage Capacity Cost (USD/capacity_units)
+    - charge_capex: Storage Charge Cost (USD/charge_units)
+
+    """
+
+    def setup(self):
+        self.config = GenericStorageCostConfig.from_dict(
+            merge_shared_inputs(self.options["tech_config"]["model_inputs"], "cost"), strict=False
+        )
+
+        super().setup()
+
+        charge_units = self.config.commodity_units
+        if self.config.commodity_units in ["W", "kW", "MW", "GW", "TW"]:
+            capacity_units = f"{self.config.commodity_units}*h"
+        else:
+            capacity_units = self.config.commodity_units.split("/")[0]
+
+        self.add_input(
+            "max_charge_rate",
+            val=self.config.max_charge_rate,
+            units=charge_units,
+            desc="Storage charge/discharge rate",
+        )
+        self.add_input(
+            "max_capacity",
+            val=self.config.max_capacity,
+            units=capacity_units,
+            desc="Storage storage capacity",
+        )
+
+    def compute(self, inputs, outputs, discrete_inputs, discrete_outputs):
+        storage_duration_hrs = 0.0
+
+        if inputs["max_charge_rate"] > 0:
+            storage_duration_hrs = inputs["max_capacity"] / inputs["max_charge_rate"]
```

Comment:
I think we need to provide an else that raises an error because if the given catch is true and storage duration hours won't be calculated and will later throw an error on line 85 that I expect will be somewhat cryptic.